### DISCUSSION TITLE
How much did it cost to deploy the contracts?

### DISCUSSION THREAD

**Discussion by cdt-eth** [nounsDAO/nouns-monorepo#398] [Solidity / smart contracts] [answered]
How much did it cost to deploy the contracts?

**Comment by solimander** [comment, marked as answer]
4.5 ETH @ 55 gwei, or roughly 81 million gas.

**Reply by Bagu-Hanto** [reply]
What aspects caused it to be so expensive? Was it certain contracts that were more expensive to deploy?

**Comment by eduardocruz** [comment]
Is the current answer of 4.5eth still correct? 
This post below, from july 2022,  which already estimated the cost down to 3.7eth, says it is now 4.2 times cheaper than 3.7eth to deploy. Is this correct, did the cost drop drastically? Can someone confirm/update the cost?
 
https://mirror.xyz/0x10072dfc23771101dC042fD0014f263316a6E400/8g7zt8354ZWlOJ5FYvtQ2LP9t0BgOaNKCw4_AEpuQEs

**Comment by eladmallel** [comment]
What was made 4.2x cheaper is the uploading of art to the descriptor, which in the original deployment was roughly 78.4% of the total gas spent (3.53 ETH of the total 4.5 ETH), meaning roughly 63,504,000 gas.
The rest of the operations were then 81M gas minus the above = 17,496,000 gas.
Let's apply the 4.2x discount on the art, and get to 15,120,000.
Add them back together: 17,496,000 + 15,120,000 = 32,616,000.

Now let's compare to the original deployment:
32,616,000 / 81,000,000 = ~40.26%.

If you deployed the original contracts with descriptor V2 and the same art, it would cost roughly 40% the original deployment in terms of gas units.
Hard to translate it to ETH you should expect due to variations in:

1. Gas cost (you can get cheaper than 55 gwei these days)
2. ETH price in fiat (we're at less than half the ETH/USD price today vs the original deployment)

Hope this is helpful!
